### PR TITLE
Add Demo of Sending Direct Message

### PR DESCRIPTION
**Pull request by joelklabo**
IMG_3909
Simulator Screenshot - iPhone 14 Pro - 2023-08-13 at 17 21 23

### CODE REVIEW

File: demo/NostrSDKDemo/Demo Views/DirectMessageDemoView.swift [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        }
+    }
+
+    private func readyToSend() -> Bool {
+        if !message.isEmpty &&
+            recipientPublicKeyIsValid &&
+            senderPrivateKeyIsValid {
+            return true
+        }
+        return false
```
Comment (bryanmontz) [inline demo/NostrSDKDemo/Demo Views/DirectMessageDemoView.swift:81]: You don't need the returns:
```suggestion
        !message.isEmpty &&
        recipientPublicKeyIsValid &&
        senderPrivateKeyIsValid
```

File: demo/NostrSDKDemo/Views/KeyInputSectionView.swift [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        }
+        .onChange(of: key) { _ in
+            isValid = checkValidity(key: key, type: type)
+        }
+        .onAppear {
+            isValid = checkValidity(key: key, type: type)
+        }
+    }
+
+    private func label(for type: KeyType) -> String {
```
Comment (bryanmontz) [inline demo/NostrSDKDemo/Views/KeyInputSectionView.swift:49]: Add this to the `KeyType` enum itself like this:
```swift
var label: String {
    switch self {
    case .public: return "Public Key"
    case .private: return "Private Key"
    }
}
```

Then the call sites are much cleaner:
```swift
keyType.label
```

File: demo/NostrSDKDemo/Views/KeyInputSectionView.swift [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+            return checkValidity(publicKey: key)
+        case .private:
+            return checkValidity(privateKey: key)
+        }
+    }
+
+    // swiftlint:disable unused_optional_binding
+    private func checkValidity(publicKey: String) -> Bool {
+        if publicKey.contains("npub") {
+            if let _ = PublicKey(npub: publicKey) {
```
Comment (bryanmontz) [inline demo/NostrSDKDemo/Views/KeyInputSectionView.swift:70]: These 5 lines reduce to:
```swift
return PublicKey(pub: publicKey) != nil
```

Can be applied to all the other cases too.

File: demo/NostrSDKDemo/Views/KeyInputSectionView.swift [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    private func label(for type: KeyType) -> String {
+        switch type {
+        case .public:
+            return "Public Key"
+        case .private:
+            return "Private Key"
+        }
+    }
+
+    private func checkValidity(key: String, type: KeyType) -> Bool {
```
Comment (bryanmontz) [inline demo/NostrSDKDemo/Views/KeyInputSectionView.swift:58]: How about:
```swift
private func isKeyValid(_ key: String, type: KeyType) -> Bool
```

**Review by bryanmontz — CHANGES_REQUESTED** [review]
Good work, Joel! Few small comments.

File: demo/NostrSDKDemo/Views/KeyInputSectionView.swift (hunk 2 above)
Comment (joelklabo) [reply demo/NostrSDKDemo/Views/KeyInputSectionView.swift]: 🤙

File: demo/NostrSDKDemo/Views/KeyInputSectionView.swift (hunk 3 above)
Comment (joelklabo) [reply demo/NostrSDKDemo/Views/KeyInputSectionView.swift:67]: Nice, was able to get rid of this too: `    // swiftlint:disable unused_optional_binding
`

File: demo/NostrSDKDemo/Views/KeyInputSectionView.swift (hunk 4 above)
Comment (joelklabo) [reply demo/NostrSDKDemo/Views/KeyInputSectionView.swift]: 👍

**Pull request merged**